packages/python/accern_xyme/accern_xyme.py

```diff
@@ -891,18 +900,69 @@ def get_kafka_error_topic(self) -> str:
         assert res is not None
         return res
 
+    def get_kafka_error_message_topic(self) -> str:
+        res = cast(KafkaTopicNames, self.request_json(
+            METHOD_GET, "/kafka_topic_names", {}))["error_msg"]
+        assert res is not None
+        return res
+
     def delete_kafka_error_topic(self) -> KafkaTopics:
         return cast(KafkaTopics, self.request_json(
             METHOD_POST, "/kafka_topics", {
                 "num_partitions": 0,
             }))
 
-    def read_kafka_errors(self, offset: str = "current") -> List[str]:
+    def read_kafka_errors(
+            self,
+            consumer_type: str,
+            offset: str = "current") -> List[str]:
+        if consumer_type not in CONSUMER_TYPES:
+            raise ValueError(
+                f"consumer_type cannot be {consumer_type} "
+                "for reading kafka errors. provide consumer type from "
+                f"{CONSUMER_TYPES}")
         return cast(List[str], self.request_json(
             METHOD_GET, "/kafka_msg", {
                 "offset": offset,
+                "consumer_type": consumer_type,
             }))
 
+    def read_kafka_full_json_errors(
+            self,
+            input_id_path: List[str]) -> Iterable[Tuple[str, Any]]:
+        errs = self.read_kafka_errors(consumer_type=CONSUMER_ERR)
+        msgs = self.read_kafka_errors(consumer_type=CONSUMER_ERR_MSG)
```

Comment:
hmm, you wanna do that in a loop maybe? -- the thing is calling read_kafka_full_json_errors multiple times will lose the state so if the err_msg for an err doesn't appear in this call but the next we won't know about the error at the second call anymore